wgpu-types/src/lib.rs

```diff
@@ -4302,73 +4302,73 @@ pub struct VertexAttribute {
+    /// Two unsigned bytes (u8). `vec2<u32>` in shaders.
     Uint8x2 = 0,
-    /// Four unsigned bytes (u8). `uvec4` in shaders.
+    /// Four unsigned bytes (u8). `vec4<u32>` in shaders.
     Uint8x4 = 1,
-    /// Two signed bytes (i8). `ivec2` in shaders.
+    /// Two signed bytes (i8). `vec2<i32>` in shaders.
     Sint8x2 = 2,
-    /// Four signed bytes (i8). `ivec4` in shaders.
+    /// Four signed bytes (i8). `vec4<i32>` in shaders.
     Sint8x4 = 3,
-    /// Two unsigned bytes (u8). [0, 255] converted to float [0, 1] `vec2` in shaders.
+    /// Two unsigned bytes (u8). [0, 255] converted to float [0, 1] `vec2<f32>` in shaders.
     Unorm8x2 = 4,
-    /// Four unsigned bytes (u8). [0, 255] converted to float [0, 1] `vec4` in shaders.
+    /// Four unsigned bytes (u8). [0, 255] converted to float [0, 1] `vec4<f32>` in shaders.
     Unorm8x4 = 5,
-    /// Two signed bytes (i8). [-127, 127] converted to float [-1, 1] `vec2` in shaders.
+    /// Two signed bytes (i8). [-127, 127] converted to float [-1, 1] `vec2<f32>` in shaders.
     Snorm8x2 = 6,
-    /// Four signed bytes (i8). [-127, 127] converted to float [-1, 1] `vec4` in shaders.
+    /// Four signed bytes (i8). [-127, 127] converted to float [-1, 1] `vec4<f32>` in shaders.
     Snorm8x4 = 7,
-    /// Two unsigned shorts (u16). `uvec2` in shaders.
+    /// Two unsigned shorts (u16). `vec2<u32>` in shaders.
     Uint16x2 = 8,
-    /// Four unsigned shorts (u16). `uvec4` in shaders.
+    /// Four unsigned shorts (u16). `vec4<u32>` in shaders.
     Uint16x4 = 9,
-    /// Two signed shorts (i16). `ivec2` in shaders.
+    /// Two signed shorts (i16). `vec2<i32>` in shaders.
     Sint16x2 = 10,
-    /// Four signed shorts (i16). `ivec4` in shaders.
+    /// Four signed shorts (i16). `vec4<i32>` in shaders.
     Sint16x4 = 11,
-    /// Two unsigned shorts (u16). [0, 65535] converted to float [0, 1] `vec2` in shaders.
+    /// Two unsigned shorts (u16). [0, 65535] converted to float [0, 1] `vec2<f32>` in shaders.
     Unorm16x2 = 12,
-    /// Four unsigned shorts (u16). [0, 65535] converted to float [0, 1] `vec4` in shaders.
+    /// Four unsigned shorts (u16). [0, 65535] converted to float [0, 1] `vec4<f32>` in shaders.
     Unorm16x4 = 13,
-    /// Two signed shorts (i16). [-32767, 32767] converted to float [-1, 1] `vec2` in shaders.
+    /// Two signed shorts (i16). [-32767, 32767] converted to float [-1, 1] `vec2<f32>` in shaders.
     Snorm16x2 = 14,
-    /// Four signed shorts (i16). [-32767, 32767] converted to float [-1, 1] `vec4` in shaders.
+    /// Four signed shorts (i16). [-32767, 32767] converted to float [-1, 1] `vec4<f32>` in shaders.
     Snorm16x4 = 15,
-    /// Two half-precision floats (no Rust equiv). `vec2` in shaders.
+    /// Two half-precision floats (no Rust equiv). `vec2<f16>` in shaders.
     Float16x2 = 16,
-    /// Four half-precision floats (no Rust equiv). `vec4` in shaders.
+    /// Four half-precision floats (no Rust equiv). `vec4<f16>` in shaders.
```

Comment:
The spec lists that `unorm`/`snorm`/`float` can all be accessed as `f16`/`f32` on the shader side.
Not sure what's best to document. All `f32`? or mention that both `f16`/`f32` work?

ref: https://gpuweb.github.io/gpuweb/#vertex-formats

Reply:
Is `f16` usable currently from WGSL? The two approaches I can think of are to match the spec and list both `f16` and `f32` for the appropriate types or match what's supported by naga. As a user, I would prefer to see a WGSL type in the docs that currently works, but I'm happy to make changes to this PR to match whatever people decide.

Reply:
`f16` is not yet implemented in naga (see https://github.com/gfx-rs/naga/issues/1884).

I don't have a strong opinion either way.

Reply:
Maybe lets use `vec4<{float}>` or something similar.

Reply:
Do vertex formats like `Float64x3` also require the WGSL type to be`vec3<f32>` on the current version? One approach is to use `f32` for all the docs for all the float formats and make a separate issue to update the documentation once `f16` and `f64` are supported in shaders.

Reply:
> Do vertex formats like `Float64x3` also require the WGSL type to be`vec3<f32>` on the current version?

The `VERTEX_ATTRIBUTE_64BIT` feature is apparently not currently advertised by any backend.

> One approach is to use `f32` for all the docs for all the float formats and make a separate issue to update the documentation once `f16` and `f64` are supported in shaders.

I think we can either do this or go with a generic solution like @cwfitzgerald was proposing. Thoughts?

Reply:
Either solution work for me

Reply:
I've added another commit to use f32 for format types represented as float in the shaders.